framework/src/utils/FillBetweenPointVectorsTools.C

```diff
@@ -466,7 +467,7 @@ surrogateGenerator(std::vector<Real> & weighted_surrogate_index,
       weighted_surrogate_index[j] += (*it_0 - unweighted_surrogate_index[j - 1]) * wt[it_start - 1];
       weighted_surrogate_index[j] +=
           (unweighted_surrogate_index[j] - *(it_1 - 1)) * wt[it_start + it_dist - 1];
-      for (unsigned int k = 1; k < it_dist; k++)
+      for (const auto k : make_range(long(1), it_dist))
```

Comment:
Here is the compiler error when I don't cast to long:
`In file included from /Users/behnpa/projects/moose_quadratic_ring/framework/build/unity_src/utils_Unity.C:20:
/Users/behnpa/projects/moose_quadratic_ring/framework/src/utils/FillBetweenPointVectorsTools.C:470:27: error: no matching function for call to 'make_range'
      for (const auto k : make_range(1, it_dist))`
I tried using auto instead of long, and got:
`/Users/behnpa/projects/moose_quadratic_ring/framework/src/utils/FillBetweenPointVectorsTools.C:470:38: error: 'auto' not allowed here
      for (const auto k : make_range(auto(1), it_dist))
`

Reply:
For loops that start from 1, should we just use the other style of for loop?

Reply:
unless you find a way to get the right data type for the 1 with various compiling options I would say yes

Reply:
could be overkill but this would work...

`make_range(castIndexForRange(foo,1),foo)`

```
template <typename T, typename P>
const T
castIndexForRange(const T &, //type_input
                  const P & value_input)
{
  return (T)(value_input);
}
```

Reply:
`make_range()` is just some syntactic sugar.  It's just to make it easier to avoid the typical C-style mistake of `for (foo i=begin(); i != expensive_call_that_shouldnt_be_here(); ++i)`, not because there's anything else we could fix with range-for vs C-style-for loops.  (though I'll bet we have a lot of both kinds of loops that should be `std::foreach`, and I'm thinking of wrapping that to play with different `std::execution::` options...)

`castIndexForRange()` is syntactic Carolina Reaper powder.  Maybe it's an acquired taste?

I think C-style is fine; otherwise `1l` is more terse than `long(1)`